marklogic-unit-test-modules/src/main/ml-modules/root/test/test-coverage.xqy

```diff
@@ -71,8 +73,8 @@ declare private function cover:_task-cancel-safe(
     for $breakpoint in dbg:breakpoints($id)
     return dbg:clear($id, $breakpoint),
     dbg:detach($id),
-    if (fn:empty(dbg:wait($id, 10))) then
-      fn:error(xs:QName("FAILED-TO-CANCEL"), "unable to cancel a debugging request")
+    if (fn:empty(dbg:wait($id, if ($TIMEOUT eq "%%mlDebugWaitTimeout%%") then $DEFAULT_TIMEOUT else $TIMEOUT)))
```

Comment:
Note that this reference will be replaced as well, so it will resolve to e.g. "if ($TIMEOUT eq 8)", and it will in fact equal 8 - which means DEFAULT_TIMEOUT will be used instead. 

I think you want to do an fn:contains here on "mlDebugWaitTimeout". 

I'm still hesitant though about having this rely on a feature of ml-gradle. How about if this depends on a server field instead?

Reply:
I'm leaning towards a server field here. In my experience, unit tests typically run against a single-node cluster, so the issue of a server field being set on one host but not another won't matter. In a 3+ node cluster, it's still simple enough to have all requests go to the same host.

Reply:
That's fair. How about if we do `%%marklogicUnitTestDebugTimeout%%`? That makes it clear it's specific to marklogic-unit-test (wish that itself had a much shorter, snappier name).

Reply:
@peetkes Just a note, I'm happy to merge this with the above changes.